cmd/bosun/expr/funcs.go

```diff
@@ -384,6 +384,161 @@ var builtins = map[string]parse.Func{
 		F:       V,
 		MapFunc: true,
 	},
+	"aggregate": {
+		Args:   []models.FuncType{models.TypeSeriesSet, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   tagFirst,
```

Comment:
This should be a function that extracts what the resulting tag keys will be, so it will come from `groups` arg to Aggregate `arg[1]`. The reason is that if you look in expr/parse, the nodes have a Tag method which makes sure at parse time that the resulting tags of function make sense in any sort of union operations.

Reply:
Ah, thanks, that makes sense now. I updated it, please have another look when you get a chance.